app/scripts/background.js

```diff
@@ -1286,24 +1284,18 @@ const addAppInstalledEvent = () => {
   }, 500);
 };
 
-// On first install, open a new tab with MetaMask
-async function onInstall() {
-  const sessionData =
-    isManifestV3 || isFirefox
-      ? await browser.storage.session.get(['isFirstTimeInstall'])
-      : await global.sessionStorage.getItem('isFirstTimeInstall');
-
-  const isFirstTimeInstall = sessionData?.isFirstTimeInstall;
-
-  if (process.env.IN_TEST) {
-    addAppInstalledEvent();
-  } else if (!isFirstTimeInstall && !process.env.METAMASK_DEBUG) {
+/**
+ * Trigger actions that should happen only upon initial install (e.g. open tab for onboarding).
+ */
+function onInstall() {
+  log.debug('First install detected');
+  addAppInstalledEvent();
```

Comment:
Previously this was never called in non-test debug builds (e.g. with `IN_TEST` is false and `METAMASK_DEBUG` is true), so this is technically a change in behavior. But it's not one that matters.

It's OK for this to be called in debug builds, it'll just use the mock Segment instance. In fact it's better this way for ease of debugging.

Reply:
I think this is now being called twice in some cases (here, and two lines down)

Reply:
Good catch, removed here: https://github.com/MetaMask/metamask-extension/pull/31332/commits/2f3b1e395f3789d2ad2182a7abd168f2604e0a61